pkg/ingester/ingester.go

```diff
@@ -2211,6 +2229,12 @@ func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
 		OutOfOrderCapMax:               i.cfg.BlocksStorageConfig.TSDB.OutOfOrderCapMax,
 		EnableOverlappingCompaction:    false, // Always let compactors handle overlapped blocks, e.g. OOO blocks.
 		EnableNativeHistograms:         i.cfg.BlocksStorageConfig.TSDB.EnableNativeHistograms,
+		BlockChunkQuerierFunc: func(b tsdb.BlockReader, mint, maxt int64) (storage.ChunkQuerier, error) {
```

Comment:
This hook seems doesn't work for the new Head and OOO querier as they bypass the hook and create their own querier instead.
https://github.com/prometheus/prometheus/blob/main/tsdb/db.go#L2177

Not a blocker for this PR.

Reply:
i will create a issue after to track.